test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,302 @@
+                ).to.be.revertedWith('invalid-array-length');
+            });
+        });
+
+        describe('Withdrawals', () => {
+            it('Should withdraw native tokens', async () => {
+                await iexecPocoAsAnyone.deposit({ value: DEPOSIT_AMOUNT_ETHER });
+
+                expect(await iexecPocoAsAnyone.callStatic.withdraw(DEPOSIT_AMOUNT)).to.be.true;
+                await expect(iexecPocoAsAnyone.withdraw(DEPOSIT_AMOUNT))
+                    .to.changeEtherBalances(
+                        [anyone, iexecPoco],
+                        [DEPOSIT_AMOUNT_ETHER, -DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(anyone.address, ZERO_ADDRESS, DEPOSIT_AMOUNT);
+
+                await checkBalances(anyone.address, BigNumber.from(0));
+            });
+
+            it('Should withdraw native tokens to another address', async () => {
+                await expect(
+                    iexecPoco.deposit({ value: DEPOSIT_AMOUNT_ETHER }),
+                ).to.changeEtherBalances(
+                    [iexecAdmin, iexecPoco],
+                    [-DEPOSIT_AMOUNT_ETHER, DEPOSIT_AMOUNT_ETHER],
+                );
+                expect(await iexecPocoAsAnyone.balanceOf(iexecAdmin.address)).to.equal(
+                    DEPOSIT_AMOUNT,
+                );
+
+                const withdrawToArgs = [DEPOSIT_AMOUNT, anyone.address] as [BigNumber, string];
+                expect(await iexecPoco.callStatic.withdrawTo(...withdrawToArgs)).to.be.true;
+                await expect(iexecPoco.withdrawTo(...withdrawToArgs))
+                    .to.changeEtherBalances(
+                        [anyone, iexecPoco],
+                        [DEPOSIT_AMOUNT_ETHER, -DEPOSIT_AMOUNT_ETHER],
+                    )
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(iexecAdmin.address, ZERO_ADDRESS, DEPOSIT_AMOUNT);
+
+                expect(await iexecPocoAsAnyone.balanceOf(iexecAdmin.address)).to.equal(0);
+            });
+
+            it('Should not withdraw native tokens with empty balance', async () => {
+                await expect(
+                    iexecPocoAsAnyone.withdraw(DEPOSIT_AMOUNT),
+                ).to.be.revertedWithoutReason();
+            });
+
+            it('Should not withdraw native tokens with insufficient balance', async () => {
+                await iexecPocoAsAnyone.deposit({ value: DEPOSIT_AMOUNT_ETHER });
+
+                await expect(
+                    iexecPocoAsAnyone.withdraw(DEPOSIT_AMOUNT.mul(2)),
+                ).to.be.revertedWithoutReason();
+            });
+        });
+
+        describe('Recover', () => {
+            it('Should recover extra balance', async () => {
+                await iexecAdmin.sendTransaction({
+                    to: iexecPoco.address,
+                    value: DEPOSIT_AMOUNT_ETHER,
+                });
+                const delta = 0;
```

Comment:
Hard to update test with `delta > 0`?

Reply:
I don't understand wen the case delta > 0 could happen ... that's mean sRLC total balance < native balance poco

Reply:
You can can deposit funds on proxy address then (re-)set `totalSupply` manually to create a delta with
```
import { setStorageAt } from '@nomicfoundation/hardhat-network-helpers';

await setStorageAt(
    proxyAddress,
    '0x0c', // Slot index of `m_totalSupply` in Store
    BigNumber.from(100).toHexString(),
);
```

Use this helper instead of low level hardhat RPC call (https://github.com/NomicFoundation/hardhat/issues/4625)

Reply:
Thanks